reference/kubectl-plugin.md

```diff
@@ -0,0 +1,219 @@
+    ```
+    {% endtab %}
+
+    {% tab title="Expected Output" %}
+    ```text
+
+     ID          GRPC ENDPOINT   STATUS
+     mayastor-2  10.1.0.7:10124  Online
+     mayastor-1  10.1.0.6:10124  Online
+     mayastor-3  10.1.0.8:10124  Online
+    ```
+    {% endtab %}
+    {% endtabs %}
+
+ {% hint style="warning" %}
+ All the above resource information can be retrieved for a particular resource using its ID. The command to do so is as follows:
+ kubectl mayastor get &lt;resource_name&gt; &lt;resource_id&gt;
+ {% endhint %}
+
+Mayastor kubectl plugin can also be used for implementing operations such as:
+
+1. Scaling up volume using Volume ID
+   {% tabs %}
+    {% tab title="Command" %}
+    ```text
+    kubectl mayastor scale volume <volume_id> <size>
+    ```
+    {% endtab %}
+
+    {% tab title="Expected Output" %}
+    ```text
+    Volume 0c08667c-8b59-4d11-9192-b54e27e0ce0f Scaled Successfully 🚀
+    ```
+    {% endtab %}
+    {% endtabs %}
+
+
+2. Retrieving resource specs in any desired format
+
+   {% tabs %}
+    {% tab title="Command" %}
+    ```text
+    kubectl mayastor -ojson get <resource_type>
+    ```
+    {% endtab %}
+
+    {% tab title="Expected Output" %}
+    ```text
+    [{"spec":{"num_replicas":2,"size":67108864,"status":"Created","target":{"node":"ksnode-2","protocol":"nvmf"},"uuid":"5703e66a-e5e5-4c84-9dbe-e5a9a5c805db","topology":{"explicit":{"allowed_nodes":["ksnode-1","ksnode-3","ksnode-2"],"preferred_nodes":["ksnode-2","ksnode-3","ksnode-1"]}},"policy":{"self_heal":true}},"state":{"target":{"children":[{"state":"Online","uri":"bdev:///ac02cf9e-8f25-45f0-ab51-d2e80bd462f1?uuid=ac02cf9e-8f25-45f0-ab51-d2e80bd462f1"},{"state":"Online","uri":"nvmf://192.168.122.6:8420/nqn.2019-05.io.openebs:7b0519cb-8864-4017-85b6-edd45f6172d8?uuid=7b0519cb-8864-4017-85b6-edd45f6172d8"}],"deviceUri":"nvmf://192.168.122.234:8420/nqn.2019-05.io.openebs:nexus-140a1eb1-62b5-43c1-acef-9cc9ebb29425","node":"ksnode-2","rebuilds":0,"protocol":"nvmf","size":67108864,"state":"Online","uuid":"140a1eb1-62b5-43c1-acef-9cc9ebb29425"},"size":67108864,"status":"Online","uuid":"5703e66a-e5e5-4c84-9dbe-e5a9a5c805db"}}]
+    ```
+    {% endtab %}
+    {% endtabs %}
+
+3. Retrieving replica topology for specific volumes.
+
+   {% tabs %}
+    {% tab title="Command" %}
+    ```text
+     kubectl mayastor get volume-replica-topology <volume_id>
+    ```
+    {% endtab %}
+
+    {% tab title="Expected Output" %}
+    ```text
+     ID                                    NODE      POOL              STATUS
+     93b1e1e9-ffcd-4c56-971e-294a530ea5cd  ksnode-2  pool-on-ksnode-2  Online
+     88d89a92-40cf-4147-97d4-09e64979f548  ksnode-3  pool-on-ksnode-3  Online
+    ```
+    {% endtab %}
+    {% endtabs %}
+
+
+{% hint style="warning" %}
+The plugin needs access to `mayastor rest server` for execution. It gets the master node IP from the kube-config file. In case of any failure, the rest endpoint can be specified using the ‘–rest’ flag.
```

Comment:
```suggestion
The plugin requires access to the `Mayastor REST server` for execution. It gets the master node IP from the kube-config file. In case of any failure, the REST endpoint can be specified using the ‘–rest’ flag.
```